app/gui/src/controller/ide/desktop.rs

```diff
@@ -48,13 +48,28 @@ pub struct Handle {
 }
 
 impl Handle {
-    /// Create a project controller handle with already loaded project model.
-    pub fn new_with_project(
+    /// Create IDE controller. If `maybe_project_name` is `Some`, a project with provided name will
+    /// be opened. Otherwise controller will be used for project manager operations by Welcome
+    /// Screen.
+    pub async fn new(
         project_manager: Rc<dyn project_manager::API>,
-        initial_project: model::Project,
+        project_name: Option<ProjectName>,
+    ) -> FallibleResult<Self> {
+        let project = match project_name {
+            Some(name) => Some(Self::init_project_model(project_manager.clone_ref(), name).await?),
+            None => None,
```

Comment:
This is a mapping, it should use map

Reply:
It is actually impossible here due to async call to `init_project_model`, as you can't use `await` in `Option::map`'s closure. There might be one other solution through `futures::stream::iter` and `StreamExt::then`, but I think it's a little overkill.